tests/robottelo/test_decorators.py

```diff
@@ -75,6 +96,78 @@ def bomb(_):
         decorators._get_bugzilla_bug = bomb
         self.assertFalse(decorators.bz_bug_is_open(self.bug_id))
 
+    def test_upstream_with_whiteboard(self):
+        """Assert that upstream bug is not affected by whiteboard texts"""
+        class MockBug(object):  # pylint:disable=R0903
+            """A mock bug"""
+            status = None
+            whiteboard = None
+        conf.properties['main.upstream'] = str(1)  # upstream mode
+        decorators._get_bugzilla_bug = lambda bug_id: MockBug()
+        # Assert bug is really closed with valid/invalid whiteboard texts
+        for MockBug.status in BZ_CLOSED_STATUSES:
+            for MockBug.whiteboard in (self.valid_whiteboard_data +
+                                       self.invalid_whiteboard_data):
+                self.assertFalse(decorators.bz_bug_is_open(self.bug_id))
+
+        # Assert bug is really open with valid/invalid whiteboard texts
+        for MockBug.status in BZ_OPEN_STATUSES:
+            for MockBug.whiteboard in (self.valid_whiteboard_data +
+                                       self.invalid_whiteboard_data):
+                self.assertTrue(decorators.bz_bug_is_open(self.bug_id))
+
+    def test_downstream_valid_whiteboard(self):
+        """Assert ``True`` is returned if
+        - bug is in any status
+        - bug has a valid Whiteboard text
+        - robottelo in downstream mode
+
+        """
+        class MockBug(object):  # pylint:disable=R0903
+            """A mock bug with a closed status."""
+            status = None
+            whiteboard = None
+        conf.properties['main.upstream'] = str(0)  # downstream mode
+        decorators._get_bugzilla_bug = lambda bug_id: MockBug()
+        for MockBug.status in BZ_OPEN_STATUSES + BZ_CLOSED_STATUSES:
+            for MockBug.whiteboard in self.valid_whiteboard_data:
+                self.assertTrue(decorators.bz_bug_is_open(self.bug_id))
+
+    def test_downstream_closedbug_invalid_whiteboard(self):
+        """Assert ``False`` is returned if
+        - bug is in closed status
+        - bug has an invalid Whiteboard text
+        - robottelo in downstream mode
+
+        """
+        class MockBug(object):  # pylint:disable=R0903
+            """A mock bug with a closed status."""
+            status = None
+            whiteboard = None
```

Comment:
This exact same class is declared in just about every single test. How about declaring this class once? Even better, how about mocking out `decorators._get_bugzilla_bug` as needed?

```python
with mock.object(decorators, '_get_bugzilla_bug'):
    ...
```